### PR TITLE
[APPC-4209] Integrate MiPay

### PR DESCRIPTION
**Pull request by giovanni-aptoide**
**What does this PR do?**

   Integrate new payment method: mipay

**Database changed?**

   No

**How should this be manually tested?**

  with a gamed injected with xiaomi oemid, try to make a payment in one of the available countries for mipay, like Russia.
  On IAP or onboarding you should have just 1 option to make payments

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-4209](https://aptoide.atlassian.net/browse/APPC-4209)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] Database migration (if applicable)?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

[APPC-4209]: https://aptoide.atlassian.net/browse/APPC-4209?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/billing/mipay/MiPayViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+  fun handleWebViewResult(result: ActivityResult) {
+    when (result.resultCode) {
+      WebViewActivity.SUCCESS -> {
+        startTransactionStatusTimer()
+      }
+
+      WebViewActivity.FAIL,
+      WebViewActivity.USER_CANCEL -> {
+        analytics.sendPaymentErrorEvent(
```
Comment (rodolfobcardoso) [inline app/src/main/java/com/asfoundation/wallet/billing/mipay/MiPayViewModel.kt:262]: On user cancelation we shouldn't send an error event, only if there was an actual error with the payment. On user cancel it should go back to the payment methods

File: app/src/main/java/com/asfoundation/wallet/onboarding_new_payment/mipay/OnboardingMiPayFragment.kt [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    binding.errorView.root.visibility = View.GONE
+    binding.pendingUserPaymentView.root.visibility = View.GONE
+    binding.completePaymentView.visibility = View.VISIBLE
+    binding.fragmentIabTransactionCompleted.iabActivityTransactionCompleted.visibility =
+      View.VISIBLE
+    binding.fragmentIabTransactionCompleted.lottieTransactionSuccess.playAnimation()
+    binding.pendingUserPaymentView.inProgressAnimation.cancelAnimation()
+  }
+
+  fun showError(message: Int?) {
```
Comment (rodolfobcardoso) [inline app/src/main/java/com/asfoundation/wallet/onboarding_new_payment/mipay/OnboardingMiPayFragment.kt:147]: There should be an error event

**Review by rodolfobcardoso — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/asfoundation/wallet/billing/CreditsRemoteRepository.java [hunk 3]
```diff
@@ -24,7 +24,7 @@ public Single<Transaction> pay(@NotNull String walletAddress, @NotNull String si
       @Nullable String productToken) {
     return remoteRepository.registerAuthorizationProof(origin, type, entityOemId, entityDomain,
         null, "appcoins_credits", walletAddress, sku, packageName, amount, payload, callback,
-        orderReference, referrerUrl, productToken);
+        orderReference, referrerUrl, productToken, signature);
```
[the remark this answers is not in the document]
Comment (giovanni-aptoide) [reply app/src/main/java/com/asfoundation/wallet/billing/CreditsRemoteRepository.java]: removed

File: app/src/main/java/com/asfoundation/wallet/billing/mipay/MiPayFragment.kt [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+      requireArguments().containsKey(CURRENCY_KEY) &&
+      requireArguments().containsKey(TRANSACTION_DATA_KEY) &&
+      requireArguments().containsKey(AMOUNT_KEY) &&
+      requireArguments().containsKey(ORIGIN_KEY)
+    ) {
+      viewModel.getPaymentLink(
+        requireArguments().getParcelable(TRANSACTION_DATA_KEY)!!,
+        (requireArguments().getSerializable(AMOUNT_KEY) as BigDecimal).toString(),
+        requireArguments().getString(CURRENCY_KEY)!!,
+        RedirectComponent.getReturnUrl(requireContext())
```
[the remark this answers is not in the document]
Comment (giovanni-aptoide) [reply app/src/main/java/com/asfoundation/wallet/billing/mipay/MiPayFragment.kt]: created RedirectUtils

File: app/src/main/java/com/asfoundation/wallet/onboarding_new_payment/mipay/OnboardingMiPayFragment.kt (hunk 2 above)
Comment (giovanni-aptoide) [reply app/src/main/java/com/asfoundation/wallet/onboarding_new_payment/mipay/OnboardingMiPayFragment.kt:165]: done

**Review by rodolfobcardoso — APPROVED** [review]

**Review by rodolfobcardoso — APPROVED** [review]

**Pull request merged**